app/views/play/CampaignView.coffee

```diff
@@ -94,10 +93,12 @@ module.exports = class CampaignView extends RootView
             if reward not in earned[group]
               console.warn 'Filling in a gap for reward', group, reward
               earned[group].push(reward)
-
     @supermodel.loadCollection(@earnedAchievements, 'achievements', {cache: false})
 
-    @listenToOnce @campaign, 'sync', @getLevelPlayCounts
+    @campaign = new Campaign({_id:@terrain})
+    @listenToOnce @campaign, 'sync', @onCampaignLoaded
+    @campaign = @supermodel.loadModel(@campaign).model
```

Comment:
JavaScript and AJAX requests don't work that way. [JavaScript runs on a single thread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop), so nothing which happens asynchronously (like [jQuery AJAX requests](http://api.jquery.com/jquery.ajax/) which has `async` true by default) can ever begin to resolve until the current call stack, including the current function, is done. There is a real chance, though, that you're attaching a listener to a [model which the supermodel throws out and instead returns the cached equivalent](https://github.com/codecombat/codecombat/blob/master/app/models/SuperModel.coffee#L38-L47).

An alternative solution would be to refactor away from supermodel's deprecated `loadModel` and `loadCollection` functions and instead use `trackRequest`, as described in [this recipe](https://github.com/codecombat/codecombat/wiki/Call-server-from-view) for [this issue](https://github.com/codecombat/codecombat/issues/3712). Then you could listen before or after, either would work. SuperModel's weird behavior is one reason for moving away from those features. However, the CampaignView and PlayLevelView are some of the only places we're actually using SuperModel's caching system, so until a better solution is devised for these use cases, we should stick to `loadModel` and `loadCollection`.

Actually, thinking it through now, the `onCampaignLoaded` doesn't get called ever if the view was given a supermodel with a loaded Campaign. This happens in the CampaignEditorView, for one, and checking it, this setup does break. Because of the weird caching here, you'll need to check if the model returned by SuperModel is loaded and, if so, call `onCampaignLoaded` directly, otherwise set up the listen. So something like this:

```
@campaign = new Campaign({_id:@terrain})
@campaign = @supermodel.loadModel(@campaign).model
if @campaign.loaded
  @onCampaignLoaded()
else
  @listenToOnce @campaign, 'sync', @onCampaignLoaded
```

I think, eventually, the solution will be to use Promises, which nicely enforce making everything consistently asynchronous, cached or not. But that's a much larger task.